Mofo EOY 2016 fundraising snippet position updates and early prep for…

ScottDowne opened the pull request:
… spectacular snippet.

ScottDowne commented:
@glogiotatidis First patch, still need to do a patch that adds a background image, but it's turning out to be a bit more tricky, so landing some stuff now to get a hear start.

I'm building these changes based on comps in the last three images from this comment, in this ticket: https://github.com/MozillaFoundation/Advocacy/issues/400#issuecomment-267887467

There is a blue, purple, and red version. I am still missing the heart images, so ignore those for now.

Examples of what this should look like along with yml props:

## Red

![red](https://cloud.githubusercontent.com/assets/197334/21335212/2d8f81ec-c62b-11e6-94ff-5d5150ea0a44.png)

```
block_button_text: 'Close'
blockable: True
clickable: False
text: 'Thanks to donations, Mozilla does a lot more than build this awesome browser. We teach people to code, fight for online privacy, and protect the web for future generations. Just a few times a year, Mozilla asks for donations. <em>If everyone reading this chipped in 3$ to help Mozilla, we could wrap this fundraiser up in under an hour!</em> Donate to Mozilla and keep the web healthy, open, and a force for good. Thank you.'
highlight_color: '#FC8383'
text_color: '#FFFFFF'
background_color: '#DB332A'
show_donation_amount_buttons: True
donation_amount_first: 20
donation_amount_second: 10
donation_amount_third: 5
donation_amount_fourth: 3
selected_button: 'donation-amount-fourth'
show_monthly_checkbox: True
button_label: 'Donate'
selected_amount_button_color: '#FC8383'
button_font_size: 'large'
```

## Blue

![blue](https://cloud.githubusercontent.com/assets/197334/21335231/53e7cb74-c62b-11e6-89c7-8f35184a9ab1.png)

```
block_button_text: 'Close'
blockable: True
clickable: False
text: 'Thanks to donations, Mozilla does a lot more than build this awesome browser. We teach people to code, fight for online privacy, and protect the web for future generations. Just a few times a year, Mozilla asks for donations. <em>If everyone reading this chipped in 3$ to help Mozilla, we could wrap this fundraiser up in under an hour!</em> Donate to Mozilla and keep the web healthy, open, and a force for good. Thank you.'
highlight_color: '#8484D8'
text_color: '#FFFFFF'
background_color: '#5C5CBC'
show_donation_amount_buttons: True
donation_amount_first: 20
donation_amount_second: 10
donation_amount_third: 5
donation_amount_fourth: 3
selected_button: 'donation-amount-fourth'
show_monthly_checkbox: True
button_label: 'Donate'
button_color: 'orange'
selected_amount_button_color: '#8484D8'
button_font_size: 'large'
```

## Purple

![purp](https://cloud.githubusercontent.com/assets/197334/21335233/5ea9a460-c62b-11e6-9123-cda120ab0d6c.png)

```
block_button_text: 'Close'
blockable: True
clickable: False
text: 'Thanks to donations, Mozilla does a lot more than build this awesome browser. We teach people to code, fight for online privacy, and protect the web for future generations. Just a few times a year, Mozilla asks for donations. <em>If everyone reading this chipped in 3$ to help Mozilla, we could wrap this fundraiser up in under an hour!</em> Donate to Mozilla and keep the web healthy, open, and a force for good. Thank you.'
highlight_color: '#8E7091'
text_color: '#FFFFFF'
background_color: '#624964'
show_donation_amount_buttons: True
donation_amount_first: 20
donation_amount_second: 10
donation_amount_third: 5
donation_amount_fourth: 3
selected_button: 'donation-amount-fourth'
show_monthly_checkbox: True
button_label: 'Donate'
selected_amount_button_color: '#8E7091'
button_font_size: 'large'
```

glogiotatidis commented:
woot woot. @ScottDowne the background image patch is going to be in this template or in a new one?

ScottDowne commented:
This one. I also made a handful of visual fixes that should apply to
standard snippet.

ScottDowne commented:
@glogiotatidis Updates, and I think it's done.

Hover on donate button should cause images to sparkle in all but the yellow version.

The current sparkle gif I have is slightly different sizes than the non animated, but for testing purposes it was sufficient, once we ship it, the design team should have fixed images. I see no reason why this should block code review :)

The page should be responsive, so if the page resizes, hearts fill the space.

I initially used background image, but there is a bug with gifs in firefox where they only animate once: https://bugzilla.mozilla.org/show_bug.cgi?id=129986

Using img fixed this, but it meant I needed to add img elements dynamically on resize :( I statically placed in html elements to fill what is a reasonable screen size for initial load so there is no possibility for a delay. I don't bother removing elements when the page gets smaller, i just add if it gets bigger.

I'm providing you yml files for the different test cases to help you test and know what we expect the snippet to do.

[snippet-yml.zip](https://github.com/mozmar/snippets/files/665624/snippet-yml.zip)

Examples of these in action:

![red-images](https://cloud.githubusercontent.com/assets/197334/21379665/c34d83b2-c71d-11e6-9a11-28ebe9308c17.png)
![blue-images](https://cloud.githubusercontent.com/assets/197334/21379664/c3482660-c71d-11e6-8c22-ebe4a7786dd2.png)
![purple-images](https://cloud.githubusercontent.com/assets/197334/21379666/c351e7cc-c71d-11e6-969d-e7167338fdb8.png)
![yellow-images](https://cloud.githubusercontent.com/assets/197334/21379667/c359c3ac-c71d-11e6-96eb-af7ac42933ee.png)

glogiotatidis commented:
In `yellow.yml` you define `special_image_fifth` but this variable does not exist anymore. Which one is more up to date?

ScottDowne commented on campaigns/mofo-eoy-2016/snippet.html:
```diff
@@ -64,6 +81,78 @@
+  }
+
+  #launcher {
+    background: hsl(0,0%,95%);
```
Yeah, need to click "show more"

This has tripped me up before too.

ScottDowne commented:
@glogiotatidis good catch on the `special_image_fifth` I caught that after I moved code from snippet-service simple server into the snippets repo and forgot to update what was in snippets.

I updated it, in a new commit so you can see what's changed. When you merge, you can feel free to squash the commits.

ScottDowne commented on campaigns/mofo-eoy-2016/snippet.html:
```diff
@@ -286,6 +387,78 @@
+          <img src="{{ background_image }}"/>
+          <img src="{{ background_image }}"/>
+          <img src="{{ background_image }}"/>
+        {% if special_image_first %}
```
Turns out the outdated code, which I just fixed in a new pull request, added the fifth image, and also moved them around a bit.

ScottDowne replied:
So their position should make more sense now...

ScottDowne commented:
@glogiotatidis ok another quick fix to when we animate the background.

Again, left as separate so easy to see changes, but should be squashed with the second commit when ready to merge.

glogiotatidis commented on campaigns/mofo-eoy-2016/snippet.html:
```diff
@@ -385,6 +624,19 @@
           }
         {% endif %}
 
+        {% if background_image_animate and background_image %}
```
This code does not work well with special images as all images get replaced by the animated image. I'll make a comment for the snippet editors, if you think that this should be fixed lets do it in another PR

ScottDowne replied:
Yeah right now animation and special images should probably not be mixed anyway, from a design pov it's prob not good.

glogiotatidis commented:
👍 Thanks Scott, happy holidays! :) 🎄 🎅 🍷

ScottDowne commented:
Does merged mean it's live?

glogiotatidis commented:
> Does merged mean it's live?

Yes!
